unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendRequest.cs

```diff
@@ -7,20 +7,30 @@ public class FriendRequest
         public string From { get; }
         public string To { get; }
         public string MessageBody { get; }
+        public FriendRequestState State { get; set; }
+        public bool HasBodyMessage => !string.IsNullOrEmpty(MessageBody);
 
-        public FriendRequest(string friendRequestId, long timestamp, string @from, string to, string messageBody)
+        public FriendRequest(string friendRequestId, long timestamp, string from, string to, string messageBody,
+            FriendRequestState state)
         {
             FriendRequestId = friendRequestId;
             Timestamp = timestamp;
-            From = @from;
+            From = from;
             To = to;
             MessageBody = messageBody;
+            State = state;
         }
 
         public bool IsSentTo(string userId) =>
             To == userId;
 
         public bool IsReceivedFrom(string userId) =>
             From == userId;
+
+        public bool IsPending() =>
+            State == FriendRequestState.Pending;
+
+        public bool IsCompleted() =>
+            State != FriendRequestState.Pending;
```

Comment:
Does this mean that the status cancelled and rejected are considered completed?

Reply:
Yes, perhaps i should remove the method, since the naming is confusing